src/main/java/Entities/User.java

```diff
@@ -4,37 +4,64 @@
 import java.util.Random;
 
 public class User {
-    private String username;
-    // maps username (key) to user ID (value) maybe not in here but somewhere else store
-    private static HashMap<String, Integer> UserIDMap;
-    private int UserID;
+    //name is whatever appears on screen (actual name)
+    //username is login credential
+    private String name;
+    private final String username;
+    private final int UserID;
     private String password;
-    private HashMap<String, Group> groups;
-    private HashMap<String, Group> applicationsList;
+    private HashMap<Integer, Group> groups;
+    private HashMap<Integer, Group> applicationsList;
     private UserPublicProfile publicProfile;
-    private class UserPublicProfile {
-        private HashMap<String, String> preferences;
-        private UserPublicProfile(){
-//            this.preferences = HashMap<String, String> ;
-        }
-    }
+    private String email;
 
-    public User(String username, String password){
+    // maps username (key) to user ID (value) maybe not in here but somewhere else store
```

Comment:
The line below initializes a HashMap<Integer, String>, so I think you just switched the order around in this doc comment.